packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,974 @@
+    const path = event.composedPath();
+    if (this && !path.includes(this)) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.hide();
+    }
+  };
+
+  private handleLabelClick() {
+    this.displayInput.focus();
+  }
+
+  private handleComboboxMouseDown() {
+    // Ignore disabled controls
+    if (this.disabled) {
+      return;
+    }
+
+    const toggleListboxOpen = () => (this.open ? this.hide() : this.show());
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    toggleListboxOpen().then(() => {
+      setTimeout(() => this.displayInput.focus({ preventScroll: true }));
+    });
+  }
+
+  private handleComboboxKeyDown(event: KeyboardEvent) {
+    if (event.key === 'Tab') {
+      return;
+    }
+
+    this.handleDocumentKeyDown(event);
+  }
+
+  private handleClearClick(event: MouseEvent) {
+    event.stopPropagation();
+    this.clearCombobox();
+    // TODO: clearify if the listbox should stay open like this or be closed
+  }
+
+  private clearCombobox() {
+    if (this.value !== '') {
+      this.value = '';
+      this.displayInput.value = '';
+      this.lastOptionValue = '';
+      this.setSelectedOption(undefined);
+      this.displayInput.focus({ preventScroll: true });
+
+      // Emit after update
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.updateComplete.then(() => {
+        this.emit('syn-clear');
+        this.emit('syn-input');
+        this.emit('syn-change');
+      });
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private preventLoosingFocus(event: MouseEvent) {
+    // Don't lose focus of the input or propagate events
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  /* eslint-disable @typescript-eslint/no-floating-promises */
+  private handleOptionClick(event: MouseEvent) {
+    const target = event.target as HTMLElement;
+    const option = target.closest('syn-option');
+    const oldValue = this.lastOptionValue;
+    if (option && !option.disabled) {
+      this.setSelectedOption(option);
+
+      // Set focus after updating so the value is announced by screen readers
+      this.updateComplete.then(() => this.displayInput.focus({ preventScroll: true }));
+
+      if (this.value !== oldValue) {
+        // Emit after updating
+        this.updateComplete.then(() => {
+          this.emit('syn-input');
+          this.emit('syn-change');
+        });
+      }
+
+      this.hide();
+      this.displayInput.focus({ preventScroll: true });
+    }
+  }
+  /* eslint-enable @typescript-eslint/no-floating-promises */
+
+  /**
+   * Selects the following or previous option.
+   *
+   * @param isNext - A boolean indicating whether to select the following option (true)
+   *                 or the previous option (false).
+   */
+  private selectNextOption(isNext: boolean) {
+    const filteredOptions = this.getAllFilteredOptions();
```

Comment:
Maybe this should also take into account that the next wanted option can be disabled, e.g.:

```html
<syn-combobox>
  <syn-option>1</syn-option>
  <syn-option disabled>2</syn-option>
  <syn-option>3</syn-option>
</syn-combobox>
```

1. When using keyboard navigation, I have to press three times to move to "3". Maybe we should skip the disabled one?
2. When clicking a disabled option, the value of the field resets to `""`

Reply:
Our select also behaves like this, that you can navigate to an disabled option with the keys. But you are right, the disabled option should definitely not be selectable, but it should happen nothing when selecting a disabled option.